workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityDetail.cs

```diff
@@ -0,0 +1,142 @@
+
+            entityName = this.Q<Label>("entity-name");
+            entityId = this.Q<Label>("entity-id");
+            componentContainer = this.Q<ScrollView>();
+        }
+
+        public void SetSelectedEntity(EntityData entityData)
+        {
+            selected = entityData;
+            Update();
+        }
+
+        public void SetWorld(World world)
+        {
+            this.world = world;
+            selected = null;
+            componentContainer.Clear();
+            visualElements.Clear();
+            Update();
+        }
+
+        public void Update()
+        {
+            if (!selected.HasValue || !world.EntityManager.Exists(selected.Value.Entity))
+            {
+                selected = null;
+                entityName.text = "No entity selected";
+                entityId.text = string.Empty;
+                return;
+            }
+
+            entityName.text = selected.Value.Metadata;
+            entityId.text = $"Entity ID: {selected.Value.EntityId}";
+
+            UpdateComponentSet();
+
+            foreach (var element in visualElements)
+            {
+                element.Update(world.EntityManager, selected.Value.Entity);
+            }
+        }
+
+        private void UpdateComponentSet()
+        {
+            using (var components = world.EntityManager.GetComponentTypes(selected.Value.Entity, allocator: Allocator.Temp))
+            {
+                var spatialComponents = components
+                    .Where(type => typeof(ISpatialComponentData).IsAssignableFrom(type.GetManagedType()))
+                    .OrderBy(type => ComponentDatabase.GetComponentId(type.GetManagedType()));
+
+                if (AreSameComponents(spatialComponents, visualElements))
+                {
+                    return;
+                }
+
+                componentContainer.Clear();
+                visualElements.Clear();
+
+                foreach (var componentType in spatialComponents)
+                {
+                    var componentElement = Cache.Get(componentType);
+                    componentContainer.Add(componentElement);
+                    visualElements.Add(componentElement);
+                }
+            }
+        }
+
+        // NOTE: This relies on the visual elements being inserted in a sorted-order, which is the case above.
+        private static bool AreSameComponents(IEnumerable<ComponentType> componentTypes,
+            IReadOnlyList<ComponentVisualElement> componentVisualElements)
+        {
+            var i = 0;
+
+            foreach (var componentType in componentTypes)
+            {
+                if (i >= componentVisualElements.Count)
+                {
+                    return false;
+                }
+
+                if (componentType != componentVisualElements[i].ComponentType)
+                {
+                    return false;
+                }
+
+                i++;
+            }
+
+            return true;
```

Comment:
If `componentVisualElements` has more elements in it than `componentTypes`, I think this will return true. Maybe pull the count check outside of the `foreach`?

Reply:
Ah nice catch - fixed!